EDU-3434: Add slots used metric (#3186)
temporalio/documentation: docs/references/sdk-metrics.mdx

### worker_task_slots_available

> The total number of Workflow Task, Activity Task, Local Activity, or Nexus execution slots that are currently available.

Reviewer: ```suggestion
The total number of Workflow, Activity, Local Activity, or Nexus Task execution slots that are currently available.
```

### worker_task_slots_available

The total number of Workflow, Activity, Local Activity, or Nexus Task execution slots that are currently available.
Use the `worker_type` key to differentiate execution slots.
The Worker type specifies an ability to perform certain tasks.
For example, Workflow Workers execute Workflow Tasks, Activity Workers execute Activity Tasks, and so forth.

- Type: Gauge
- Available in: Core, Go, Java
- Tags: `namespace`, `task_queue`, `worker_type`

### worker_task_slots_used

> The total number of Workflow Task, Activity Task, Local Activity, or Nexus execution slots that are currently used.

Reviewer: ```suggestion
The total number of Workflow, Activity, Local Activity, or Nexus Tasks execution slots in current use.
```

Use the `worker_type` key to differentiate execution slots.

> (Workflow Workers execute Workflow Tasks; Activity Workers execute Activity Tasks, etc.)

Reviewer: ```suggestion
The Worker type specifies an ability to perform certain tasks.
For example, Workflow Workers execute Workflow Tasks, Activity Workers execute Activity Tasks, and so forth.
```
